pkg/webhook/clusterset/validatingWebhook.go

```diff
@@ -46,18 +57,48 @@ var agentClusterOwnerRef = metav1.OwnerReference{
 	APIVersion: "capi-provider.agent-install.openshift.io/v1alpha1",
 }
 
-const (
-	clusterSetLabel = "cluster.open-cluster-management.io/clusterset"
-	defaultSetName  = "default"
-)
-
 // validateResource validate:
 // 1. allow requests if the user has managedClusterSet/join all resources permission.
 // 2. if user has permission to create/update resources.
 func (a *AdmissionHandler) validateResource(request *v1.AdmissionRequest) *v1.AdmissionResponse {
 	status := &v1.AdmissionResponse{
 		Allowed: true,
 	}
+
+	//Only allow create/update global clusterset and legacy clusterset
+	if request.Resource == managedClusterSetsGVR {
+		//Only handle clusterset create/update request
+		if request.Operation != v1.Create && request.Operation != v1.Update {
+			return status
+		}
+
+		clusterset := &clusterv1beta1.ManagedClusterSet{}
+		if err := json.Unmarshal(request.Object.Raw, clusterset); err != nil {
+			status.Allowed = false
+			status.Result = &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: err.Error(),
+			}
+			return status
+		}
+		//allow create/update legacy clusterset
+		if clusterset.Spec.ClusterSelector.SelectorType == clusterv1beta1.LegacyClusterSetLabel {
+			return status
+		}
+
+		//allow create/update global clusterset
+		if clusterset.Name == clustersetutils.GlobalSetName && equality.Semantic.DeepEqual(clusterset.Spec, clustersetutils.GlobalSet.Spec) {
+			return status
+		}
+
+		status.Allowed = false
+		status.Result = &metav1.Status{
+			Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+			Message: fmt.Sprintf("Do not allow to create/update this kind of clusterset. %v", clusterset),
```

Comment:
the message seems too much, do not put clusterset payload in the message

Reply:
done